libs/sdk-backend-spi/src/backend/index.ts

```diff
@@ -117,6 +118,12 @@ export interface IAnalyticalBackend {
      */
     deauthenticate(): Promise<void>;
 
+    /**
+     * Returns a domain available on this backend.
+     * @param domain - domain ref
+     */
+    domain(domain: ObjRef): IDomain;
```

Comment:
Does ObjRef make sense here? Because according to the arch doc, you can either use URI for the domain or project id for project, right?. So maybe the ObjRef is not the best type here, we might be better with some custom type that is more explicit, like

```ts
type DomainScopeRef = { uri: string } | { workspace: string; }
```

What do you think?

Reply:
From my understanding it is always URI, so for project it will be URI as well. I will double verify this. My plan to get value for domain is from user bootstrap response which should be ref imho. 

https://docs.google.com/document/d/11H6fKlYHSCHjb7FpFUvZDCbmfLo7PzllGRjjRC2I6_8/edit#

Reply:
I found out that domain, workspace, and user are not metadata object and are only records in C4. Therefore, string URI is enough. In this case, based on Lupko suggestion, only organizationId (name used for _domain_ in Tiger is _organization_ and it will probably the name that will be used in future for domain as well) is param of the API and URI is built in bear implementation of the service to hide this detail from the higher level of abstraction.